cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1644,7 +1656,11 @@ private void interBrokerMoveReplicas() throws InterruptedException, ExecutionExc
             .collect(Collectors.toList());
         inProgressTasks.addAll(inExecutionTasks());
 
-        throttleHelper.clearThrottles(completedTasks, inProgressTasks);
+          // Clear throttles only after all tasks are complete for the brokers involved
+          if (partitionsToMove == 0) {
```

Comment:
I think you may want to check all of these conditions: 

(partitionsToMove == 0 && inExecutionTasks().isEmpty()) ||  (_stopSignal.get() != NO_STOP_EXECUTION)

Reply:
Okay 👍